api/server/models/user.js

```diff
@@ -0,0 +1,121 @@
+var GENERIC = require('../modules/generic');
+var lodash = require('lodash');
+
+module.exports = function (User) {
+    User.register_login = function (action, action_type, social_id, platform, device_id, token, email_id, name, password, currentTimestamp, currentDate, currentDateTimeDay, callback) {
+        name = name.toLowerCase();
+        email_id = email_id.toLowerCase();
+        where = {
+            email_id: email_id
+        };
+        User.find({where: where}, function (err, result) {
+            if (err) {
+                callback(null, 0, err, {});
+            } else {
+                if (action == 'login_register') {
+                    var resultSize = lodash.size(result);
+                    if (resultSize > 0) {
+                        result = result[0];
+
+                        var r_verification_status = result.verification_status;
+
+                        if (action_type == 'facebook' || action_type == 'google') {
+                            var data = {
+                                user_id: result.id
+                            };
+                            callback(null, 1, 'Success login', data);
+                        } else if (action_type == 'manual_register') {
+                            callback(null, 0, 'Email id already exists.', data);
+                        } else if (action_type == 'manual_login') {
+                            var exist_password = result.password;
+                            GENERIC.match_encode_password(password, exist_password, function (check) {
```

Comment:
in previous pull request i had ask to change GENERIC to UTIL this is not done yet